lib/hawkular/alerts/alerts_api.rb

```diff
@@ -20,6 +20,13 @@ def initialize(entrypoint = 'http://localhost:8080/hawkular/alerts', credentials
       super(entrypoint, credentials, options)
     end
 
+    # Return version and status information for the used version of Hawkular-Alerting
+    # @return [Hash{String=>String}]
+    #         ('Implementation-Version', 'Built-From-Git-SHA1', 'status')
+    def fetch_version_and_status
```

Comment:
> I am ok with fetch_version_and_status. Perhaps, it's not an idiomatic Ruby, but at least it says, it'll actually does something (fetch -> do a rest call).

the majority of the methods in client are actually doing a call, their names are not implying that and it's an implementation detail.

> 
> I assume these methods can be used in the future versions of gem, when the different versions of the h-services REST API will require different treatment by the client. Potentially frequently (depends on the impl). To do that, we can actually store the version and/or sha as a client attributes, perhaps somehow lazily (do the actual fetch, if it's firstly asked for them).
> 
> So what about having fetch_version_and_status for each client and + the top-lvl client that can call it on the all sub-clients and returns the Hash of Hash of versions. And then version field on all the clients that will represent a cached value of the version + does the lazy fetch logic. wdyt?
> 
> This way, this PR can be merged and the fetch_version_and_status will be on all the clients (actually, it still needs to be added to the inventory client) => consistency.
> And we can later add the version or perhaps backend_version fields that will expose the cached value => adding new stuff won't break the API